File: LPAv/Trabalho 9 - Python/questao1.py
import re

stringDeEntrada = input()
while (stringDeEntrada != "####"):
    if (re.match("^([0-9]{7,7}-[0-9])$", stringDeEntrada)):
        print("SIM")
    else:
        print("NAO")
    stringDeEntrada = input()

File: LPAv/Trabalho 9 - C++/questao1.cpp
#include <iostream>
#include <stdio.h>
#include <string>
#include <regex>
using namespace std;

int main()
{
    string expressaoRegular;
    cin >> expressaoRegular;
    while (expressaoRegular != "####")
    {
        if (regex_match(expressaoRegular, regex("[0-9]{7}-[0-9]{1}")))
        {
            printf("SIM\n");
        }
        else
        {
            printf("NAO\n");
        }
        cin >> expressaoRegular;
    }
}